src/utils/response.utils.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Defines reusable responses
+ */
+export default class {
+  /**
+   * Defines the specification for the "unauthorized error" response cases
+   * @param {ServerResponse} res
+   * @param {object} error
+   * @param {number} code
+   * @returns {ServerResponse} response
+   */
+  static getUnauthorizedErrorResponse(res, error, code) {
+    return res.status(code || 401).json({
+      status: 'error',
+      error
+    });
+  }
+
+  /**
+   * Defines the specification for the "success" response cases
+   * @param {ServerResponse} res
+   * @param {object} data
+   * @param {number} code
+   * @returns {ServerResponse} response
+   */
+  static getSuccessResponse(res, data, code) {
+    return res.status(code || 200).json({
+      status: 'success',
+      data
+    });
+  }
+
+  /**
+   * @param {ServerResponse} res
+   * @param {object|string} error
+   * @returns {ServerResponse} response
+   */
+  static getInternalServerErrorResponse(res, error) {
```

Comment:
I would rather we handled `500` errors on the application level like so:

```

app.use((err, req, res, next) => {
    res.status(500).json({
        status: 'error',
        error: 'Internal server error'
    });
});

```

Then do this to pass errors down to the application:

```

try {
....
} catch (error) { next(error) }

```

Reply:
OK. However, I don't remember returning an internal server error on my feature (I'll check again though). These may be from @chiomadans1759's implementation as I didn't want to tamper with her logic even while making changes to the files

Reply:
Alright then.

Reply:
@O-Osahon You've been on Academind for too long. Smiles !!!

Reply:
Lol